ladybug/_datacollectionbase.py

```diff
@@ -0,0 +1,517 @@
+# coding=utf-8
+"""Base class for Ladybug Data Collections."""
+
+from __future__ import division
+
+from .header import Header
+
+from string import ascii_lowercase
+import math
+import sys
+if (sys.version_info >= (3, 0)):
+    xrange = range
+
+
+class BaseCollection(object):
+    """Base class for all Data Collections."""
+
+    __slots__ = ('_header', '_values', '_datetimes')
+
+    def __init__(self, header, values, datetimes):
+        """Initialize base collection.
+
+        Args:
+            header: A Ladybug Header object.
+            values: A list of values.
+            datetimes: A list of Ladybug DateTime objects that aligns with
+                the list of values.
+        """
+        assert isinstance(header, Header), \
+            'header must be a Ladybug Header object. Got {}'.format(type(header))
+        assert isinstance(datetimes, (list, tuple)), \
```

Comment:
I wasn't aware that we wanted to support assigning iterators.  As of now, assigning an iterator like xrange would break certain features, like the __del__ feature.  If we want to support all iterators, I can just cast everything to a list before we assign it to the object.  Let me know if this is what you mean and I would be happy to implement it.

Reply:
Also, wouldn't this make it ok for people to assign a dictionary? Is that what we want?

Reply:
This has now bee addressed after our conversation.